### DISCUSSION TITLE
Files required by all configurations of UFS?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [ufs-community/ufs-weather-model#2300] [Q&A] [answered]
Is there a list of files that are required by all configurations of UFS, assuming that there are such files? Is [this list](https://ufs-weather-model.readthedocs.io/en/develop/InputsOutputs.html#model-configuration-files) of files complete?

**Comment by gspetro-NOAA** [comment]
@ulmononian and I agree that this list seems to capture most of the files that would be required for configuration. I'm not sure whether every configuration uses all of these files. @jkbk2004 should be able to confirm and indicate whether some configurations may require more or fewer files. @ulmononian pointed out that the regression testing framework typically generates the required files, which get populated in the case run directory, so it is a good reference when running similar cases.

**Reply by benjamin-cash** [reply]
Is there an easy way to harvest from the regression tests which files are being copied into place? I was also under the impression that not every configuration has a regression test - for example, it seems like the last version of the coupled model with a defined test is p8.

I think practically every configuration of the model requires more than these files, but those files are specific to the set of components chosen. A more precise question would be - is there any configuration of the UFS that does **not** require `diag_table`, `field_table`, `model_configure`, and `ufs.configure`?

**Comment by RatkoVasic-NOAA** [comment]
@benjamin-cash
- that basically covers all needed files
- there are no applications that do not need mentiones files (diag_table, field_table, model_configure and ufs.configure)_
- here is example on Frontera (just regional run, but I can create global case as well):
/work2/03350/rvasic/frontera/rvasic/FV3_RT/rt_46430/regional_control_intel

**Reply by benjamin-cash** [reply, marked as answer]
Thanks @RatkoVasic-NOAA ! FYI, @christinaholtNOAA @maddenp-noaa

**Comment by christinaholtNOAA** [comment]
Is it true that all the considered configurations run with FV3? Are there supported configurations that do not run with FV3 as a component and just use data for the atmospheric forcing? Do those configurations still need those files?

**Reply by RatkoVasic-NOAA** [reply]
@christinaholtNOAA FV3 is the only dynamical core currently in ufs-weather-model. Which configurations do we have that do not use dynamical core?

**Reply by benjamin-cash** [reply]
There are at least a few supported configurations that do not run with FV3, but use a data atmosphere: https://ufs-weather-model.readthedocs.io/en/develop/InputsOutputs.html#id23

My understanding is that this list of files is required even in the case where FV3 is not in use.

**Reply by benjamin-cash** [reply]
@RatkoVasic-NOAA - There are at least a couple of configurations that use a data atmosphere in place of fv3. This is a particularly useful capability for hierarchical testing, which if Mike Ek was on this thread he would enthusiastically jump in to remind us of. :)

**Reply by RatkoVasic-NOAA** [reply]
I see, there are two applications that use DATM. Thanks. Yes, those files are still needed for geometry and number of tasks (model_configure), list of output fields (diag_table, field_table), ...
